featurebyte/query_graph/sql/online_store_compute_query.py

```diff
@@ -306,6 +306,16 @@ def get_online_store_precompute_queries(
     -------
     list[OnlineStoreComputeQueryModel]
     """
-    universe_plan = OnlineStorePrecomputePlan(graph, node, adapter=get_sql_adapter(source_type))
+    # Prune the graph to only include the aggregations that are required. This is necessary when
+    # this function is called from FeatureModel._add_tile_derived_attributes() root validator
+    # because the graph may not have been pruned yet.
```

Comment:
If i understood this correctly, this is only used by saved feature, right? If that is the case, can we assume that it is pruned? This is the final constructor call just before the feature is saved to the persistent (https://github.com/featurebyte/featurebyte/blob/main/featurebyte/service/feature.py#L84-L87).

Reply:
Yes, this is used by saved feature only. I was debugging and found that the graph in the `FeatureModel`'s root validator is not pruned, but maybe that is not the final `FeatureModel` that will be saved. Will try to simplify using that assumption.

Reply:
Will create a follow up PR to address this since it requires further changes in a few places.